Radio HAT board v1.0

Akshat412 opened the pull request:
Finalized board layout and routing for the Radio HAT board.

Akshat412 committed:
Initial schematic design for RFM98 HAT

Akshat412 committed:
Finished schematic, started board layout

Akshat412 committed:
Finished board design

NeilKhera commented:
Great work! A minor thing, I think you should create a new footprint and symbol library for this board instead of using the one from PyCubed.

Schematic:
1. Mention (in a comment) the nominal VCC voltage (12V) and the max/min voltage range that the LDO can accept at its input.
2. Good job selecting standard resistor values for setting the LDO output voltage. Might be useful to mention that those resistors should have tight tolerances <=1%.
3. In the page settings of the schematic - add a board revision number (v1.0), date, title (Argus-1 Groundstation), and feel free to add your name as well.

PCB layout:
1. Set up the design rule constraints in the board design page like so. 
![image](https://github.com/cmu-spacecraft-design-build-fly-2023/Ops-Groundstation/assets/32021246/a183bd76-d748-4abd-8ad8-d29a246d387e)

You should have about 16 new errors, all relating to the annular rings on the mounting hole pad vias. You can ignore these since the annular ring is not relevant for these vias.

2. You can probably put the JLCJLC text somewhere less conspicuous - maybe within the radio footprint so that it is hidden once assembled - or along the bottom edge of the board if you'd prefer for it to be on the back. 

3. Hide the J1 mounting hole silkscreen.
4. Change "Carnegie Mellon ECE" to "Carnegie Mellon University"
5. Add a revision number silkscreen to the board (v1.0)
6. Feel free to add your name somewhere on the board if you'd like

7. I'm a little concerned about the heat produced by the LDO, since it appears the big thermal pad is not GND but rather +6V. I think it might be better to do a larger copper fill for the +6V line to aid heat dissipation. Alternatively, swap this component for a different LDO that has a grounded thermal pad.

8. The RF line should be sized for 50ohm impedance. This is only really crucial once we exceed 10% of the wavelength of the frequency we are operating on (so about 7cm), but it good practice nonetheless. You can use this calculator to determine that (https://www.everythingrf.com/rf-calculators/microstrip-impedance-calculator) which should also give you intuition about why multi-layer boards are better for RF. We don't really care too much about this so 2-layer is fine. Using the OSHPark stackup, I get a required trace thickness of about 105 mils, which is too wide to fit the SMA connector - so just make it as wide as possible (80-85 mils is possible) and it should be fine.

9. Assign 3D models to every part (the radio and sma are currently missing them). Make a folder in the Libraries for 3D Models. 
10. Export a STEP file for the board
11. Install the Interactive HTML Bom Plugin (https://github.com/openscopeproject/InteractiveHtmlBom) and export a BOM

Akshat412 committed:
Implemented changes from PR review

Changes implemented:
 - added schematic documentation and page settings
 - fixed PCB silkscreen
 - added board rev. and information on back silkscreen
 - added 3D models, generated 3D model for board
 - generated HTML BOM

NeilKhera commented:
Looks great! Just a couple more things that might be a good idea (that slipped my mind earlier)
1. Keep in mind that the 40-pin connector will be attached to the bottom of the board. 
2. It would be good to add test points on the input voltage, ldo voltage, and GND. Add silkscreen label for each test point.
3. Do both of the above and re-export the step model

Akshat412 committed:
Implemented changes from PR

Fixed 3D models of board, added testpoints on board.

Akshat412 committed:
Reworked GND stitching vias

NeilKhera reviewed (approved):
LGTM!
